core/io/resource.cpp

```diff
@@ -37,6 +37,7 @@
 #include "core/object/script_language.h"
 #include "core/os/os.h"
 #include "scene/main/node.h" //only so casting works
+#include "scene/resources/packed_scene.h"
```

Comment:
`core` shouldn't depend on `scene`. The above one with `node.h` is tech debt that needs to be solved eventually.

Reply:
I would guess that they would prefer to use a hard-coded string in this instance. The only other "safe" alternative would be to use some other sort of wrapper logic present in the scene context to handle the implementation of this function. Not sure if that would be preferred though since it ultimately makes the code more complicated - and there aren't any special cases beyond Script and PackedScene, so the effort of making it loosely coupled *and* statically typed probably isn't worth it. My thought anyway.

Reply:
Alright I've removed the dependency